### PR TITLE
MAGECLOUD-3159: Improve git installation flow

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description

Added possibility to install Magento from git branches with packages that have dependencies on Magento modules.

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3159

### Manual testing scenarios

https://wiki.corp.magento.com/display/MCP/ECE+Tools+-+Simple+Flow+to+Deploy+Magento+to+ECE+from+GitHub+Branch

Installation from git with Fastly module

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by shiftedreality — DISMISSED** [review]

**Review by bbatsche — DISMISSED** [review]

**Comment by andriyShevtsov** [comment]
Installation with 2.3-develop should be fixed

File: src/Test/Unit/Command/Dev/UpdateComposer/_files/clear_module_requirements.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    if (isset($repoOptions['type']) && $repoOptions['type'] == 'single-package') {
+        clearRequirements($repoDir);
+        continue;
+    }
+
+    foreach (glob($repoDir . '/app/code/Magento/*') as $moduleDir) {
+        clearRequirements($moduleDir);
+    }
+}
```
Comment (NadiyaS) [inline src/Test/Unit/Command/Dev/UpdateComposer/_files/clear_module_requirements.php]: missed new line?

Comment (oshmyheliuk) [reply src/Test/Unit/Command/Dev/UpdateComposer/_files/clear_module_requirements.php]: added.

**Review by NadiyaS — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**